### DISCUSSION TITLE
Multiplication of variables in math opt?

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#4112] [Feature requests] [answered]
Hello,
  will be multiplication of variables (x * y) supported in new api math_opt? I am interested in integer variables and to be able to express a constraint with multiplication. Since I was able to use it in ampl language with scip solver it should work for scip and probably also gurobi solver but I need api :-)

Thanks,
Jan

**Comment by lperron** [comment]
it works in CP-SAT if your problem is integral.

**Comment by lperron** [comment, marked as answer]
To answer your question, it is supported in C++. Not yet in python or java.

**Reply by rma350** [reply]
Quadratic objectives are supported in python, just not quadratic constraints.

Long term, feature parity between C++ and non C++ languages is coming, we are going to do something systematic so we can reasonably add new modeling features without writing a ton of code in every language.

**Comment by gregy4** [comment]
I know, I already created a model in cp-sat solver. The problem is high variation of solving time - once it is 5s, second time 10minutes. Since results are far away from lower bounds (5000 vs 70000) I’m not sure about quality of a solution. The problem is quite small, about 300 variables. I wanted to give a try to another solver.
